beacon_node/store/src/hot_cold_store.rs

```diff
@@ -80,6 +78,46 @@ pub struct HotColdDB<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> {
     _phantom: PhantomData<E>,
 }
 
+#[derive(Debug)]
+struct BlockCache<E: EthSpec> {
```

Comment:
Put the block and blob cache behind the same mutex to get rid of any deadlock issues